### PR TITLE
3.5 - Integrating latest from neo4j/3.5

### PR DESCRIPTION
**Pull request by bradnussbaum**

### CODE REVIEW

File: src/main/java/apoc/trigger/Trigger.java [hunk 1]
```diff
@@ -290,9 +341,9 @@ private void executeTriggers(TransactionData txData, String phase) {
                             log.error( "Failed to get last transaction id: " + e.getMessage() );
                         }
                     }
-                    if( ( (Map<String,Object>) data.get( "config" )).get( "params" ) != null)
```
Comment (ThomasMatlak) [inline src/main/java/apoc/trigger/Trigger.java]: Could removing the intermediate `config` key here be part of the problem?
Noting as something to look into.

Comment (bradnussbaum) [reply src/main/java/apoc/trigger/Trigger.java]: I noticed that `config` was removed in place of `params`. Any idea why one was done over the other? If we intentionally had this as `config` to align with other usages then maybe I just need to revert that part.

Comment (ThomasMatlak) [reply src/main/java/apoc/trigger/Trigger.java]: I'm not sure of a reason for the change (might just be more semantically meaningful).
I think I know why it's failing though.

Looking at the bamboo logs, several (all? I didn't check for all of them) of the failing tests are encountering a NPE on line [378](https://github.com/graphfoundation/ongdb-apoc/blob/72340bd9c9382e2250eae4ce223049bec8c257bf/src/main/java/apoc/trigger/Trigger.java#L378).
```List<String> uidKeys = (List<String>) config.getOrDefault( "uidKeys", Collections.emptyList() );```

The variable `config` is an argument of that function, and when the function is called on line [329](https://github.com/graphfoundation/ongdb-apoc/blob/72340bd9c9382e2250eae4ce223049bec8c257bf/src/main/java/apoc/trigger/Trigger.java#L329), `data.get( "config" )` is probably `null`.

Comment (ThomasMatlak) [reply src/main/java/apoc/trigger/Trigger.java]: I did check all the `TriggerTest`s, and they're all failing from a NPE on the line in my previous comment.

Comment (bradnussbaum) [reply src/main/java/apoc/trigger/Trigger.java]: Should I keep `config` then and ditch the incoming changes that are making it `params`?

Comment (ThomasMatlak) [reply src/main/java/apoc/trigger/Trigger.java]: Just from a search of `Trigger.java` for `config`, line 329 looks to be the only one where the switch from `config` to `params` was missed. It just happened to not conflict with any of the changes being merged.

So changing `config.get( "config" )` to `config.get( "params" )` on line 329 should fix this error.

Comment (alexiudice) [reply src/main/java/apoc/trigger/Trigger.java]: @bradnussbaum @ThomasMatlak Just for context I'm pretty sure we changed the `params` to `config` to distinguish between trigger configuration attributes and actual parameters being used within the trigger query.

**Pull request merged**